### PR TITLE
update to remove object3d mesh when component removed from entity

### PR DESCRIPTION
**Pull request by kfarr**
This PR is just a simple `remove` function that removes the cursor mesh when the component is removed from an entity. I also updated the dist to reflect these changes.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -116,6 +115,14 @@ AFRAME.registerComponent('controller-cursor', {
   },
 
   /**
+   * Remove mesh.
+   */
+  remove: function () {
+    var cursorEl = this.el;
+    cursorEl.removeObject3D('cursormesh');
```
Comment (bryik) [inline index.js:122]: Does `remove` detach event listeners? I think you might have to call `pause` manually.

**Review by bryik — CHANGES_REQUESTED** [review]

**Comment by bryik** [comment]
Also, kudos for working on this despite all the examples being broken with A-Frame 0.5.0! Maybe I should bump the dependency to A-Frame master as a stop-gap.

File: index.js (hunk 1 above)
Comment (kfarr) [reply index.js:122]: Good question, it looks like `pause` is called by a-frame itself prior to `remove`. I tested this only with console log comments (it showed both pause and remove functions running when the component is removed), however I have not explicitly tested to see if the event listeners are still running upon removing the component.

Comment (bryik) [reply index.js:122]: > it looks like pause is called by a-frame itself prior to remove

Yes you're right, my mistake!

> I have not explicitly tested to see if the event listeners are still running

I'm going to trust that they are removed :smile:

**Comment by bryik** [comment]
Thanks!

**Pull request merged**